// @drdgvhbh/postgres-error-codes/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PG_SUCCESSFUL_COMPLETION = '00000';
exports.PG_WARNING = '01000';
exports.PG_DYNAMIC_RESULT_SETS_RETURNED = '0100C';
exports.PG_IMPLICIT_ZERO_BIT_PADDING = '01008';
exports.PG_NULL_VALUE_ELIMINATED_IN_SET_FUNCTION = '01003';
exports.PG_PRIVILEGE_NOT_GRANTED = '01007';
exports.PG_PRIVILEGE_NOT_REVOKED = '01006';
exports.PG_STRING_DATA_RIGHT_TRUNCATION = '01004';
exports.PG_DEPRECATED_FEATURE = '01P01';
exports.PG_NO_DATA = '02000';
exports.PG_NO_ADDITIONAL_DYNAMIC_RESULT_SETS_RETURNED = '02001';
exports.PG_SQL_STATEMENT_NOT_YET_COMPLETE = '03000';
exports.PG_CONNECTION_EXCEPTION = '08000';
exports.PG_CONNECTION_DOES_NOT_EXIST = '08003';
exports.PG_CONNECTION_FAILURE = '08006';
exports.PG_SQLCLIENT_UNABLE_TO_ESTABLISH_SQLCONNECTION = '08001';
exports.PG_SQLSERVER_REJECTED_ESTABLISHMENT_OF_SQLCONNECTION = '08004';
exports.PG_TRANSACTION_RESOLUTION_UNKNOWN = '08007';
exports.PG_PROTOCOL_VIOLATION = '08P01';
exports.PG_TRIGGERED_ACTION_EXCEPTION = '09000';
exports.PG_FEATURE_NOT_SUPPORTED = '0A000';
exports.PG_INVALID_TRANSACTION_INITIATION = '0B000';
exports.PG_LOCATOR_EXCEPTION = '0F000';
exports.PG_INVALID_LOCATOR_SPECIFICATION = '0F001';
exports.PG_INVALID_GRANTOR = '0L000';
exports.PG_INVALID_GRANT_OPERATION = '0LP01';
exports.PG_INVALID_ROLE_SPECIFICATION = '0P000';
exports.PG_DIAGNOSTICS_EXCEPTION = '0Z000';
exports.PG_STACKED_DIAGNOSTICS_ACCESSED_WITHOUT_ACTIVE_HANDLER = '0Z002';
exports.PG_CASE_NOT_FOUND = '20000';
exports.PG_CARDINALITY_VIOLATION = '21000';
exports.PG_DATA_EXCEPTION = '22000';
exports.PG_ARRAY_SUBSCRIPT_ERROR = '2202E';
exports.PG_CHARACTER_NOT_IN_REPERTOIRE = '22021';
exports.PG_DATETIME_FIELD_OVERFLOW = '22008';
exports.PG_DIVISION_BY_ZERO = '22012';
exports.PG_ERROR_IN_ASSIGNMENT = '22005';
exports.PG_ESCAPE_CHARACTER_CONFLICT = '2200B';
exports.PG_INDICATOR_OVERFLOW = '22022';
exports.PG_INTERVAL_FIELD_OVERFLOW = '22015';
exports.PG_INVALID_ARGUMENT_FOR_LOGARITHM = '2201E';
exports.PG_INVALID_ARGUMENT_FOR_NTILE_FUNCTION = '22014';
exports.PG_INVALID_ARGUMENT_FOR_NTH_VALUE_FUNCTION = '22016';
exports.PG_INVALID_ARGUMENT_FOR_POWER_FUNCTION = '2201F';
exports.PG_INVALID_ARGUMENT_FOR_WIDTH_BUCKET_FUNCTION = '2201G';
exports.PG_INVALID_CHARACTER_VALUE_FOR_CAST = '22018';
exports.PG_INVALID_DATETIME_FORMAT = '22007';
exports.PG_INVALID_ESCAPE_CHARACTER = '22019';
exports.PG_INVALID_ESCAPE_OCTET = '2200D';
exports.PG_INVALID_ESCAPE_SEQUENCE = '22025';
exports.PG_NONSTANDARD_USE_OF_ESCAPE_CHARACTER = '22P06';
exports.PG_INVALID_INDICATOR_PARAMETER_VALUE = '22010';
exports.PG_INVALID_PARAMETER_VALUE = '22023';
exports.PG_INVALID_REGULAR_EXPRESSION = '2201B';
exports.PG_INVALID_ROW_COUNT_IN_LIMIT_CLAUSE = '2201W';
exports.PG_INVALID_ROW_COUNT_IN_RESULT_OFFSET_CLAUSE = '2201X';
exports.PG_INVALID_TIME_ZONE_DISPLACEMENT_VALUE = '22009';
exports.PG_INVALID_USE_OF_ESCAPE_CHARACTER = '2200C';
exports.PG_MOST_SPECIFIC_TYPE_MISMATCH = '2200G';
exports.PG_NULL_VALUE_NOT_ALLOWED = '22004';
exports.PG_NULL_VALUE_NO_INDICATOR_PARAMETER = '22002';
exports.PG_NUMERIC_VALUE_OUT_OF_RANGE = '22003';
exports.PG_STRING_DATA_LENGTH_MISMATCH = '22026';
exports.PG_DATA_STRING_DATA_RIGHT_TRUNCATION = '22001';
exports.PG_SUBSTRING_ERROR = '22011';
exports.PG_TRIM_ERROR = '22027';
exports.PG_UNTERMINATED_C_STRING = '22024';
exports.PG_ZERO_LENGTH_CHARACTER_STRING = '2200F';
exports.PG_FLOATING_POINT_EXCEPTION = '22P01';
exports.PG_INVALID_TEXT_REPRESENTATION = '22P02';
exports.PG_INVALID_BINARY_REPRESENTATION = '22P03';
exports.PG_BAD_COPY_FILE_FORMAT = '22P04';
exports.PG_UNTRANSLATABLE_CHARACTER = '22P05';
exports.PG_NOT_AN_XML_DOCUMENT = '2200L';
exports.PG_INVALID_XML_DOCUMENT = '2200M';
exports.PG_INVALID_XML_CONTENT = '2200N';
exports.PG_INVALID_XML_COMMENT = '2200S';
exports.PG_INVALID_XML_PROCESSING_INSTRUCTION = '2200T';
exports.PG_INTEGRITY_CONSTRAINT_VIOLATION = '23000';
exports.PG_RESTRICT_VIOLATION = '23001';
exports.PG_NOT_NULL_VIOLATION = '23502';
exports.PG_FOREIGN_KEY_VIOLATION = '23503';
exports.PG_UNIQUE_VIOLATION = '23505';
exports.PG_CHECK_VIOLATION = '23514';
exports.PG_EXCLUSION_VIOLATION = '23P01';
exports.PG_INVALID_CURSOR_STATE = '24000';
exports.PG_INVALID_TRANSACTION_STATE = '25000';
exports.PG_ACTIVE_SQL_TRANSACTION = '25001';
exports.PG_BRANCH_TRANSACTION_ALREADY_ACTIVE = '25002';
exports.PG_HELD_CURSOR_REQUIRES_SAME_ISOLATION_LEVEL = '25008';
exports.PG_INAPPROPRIATE_ACCESS_MODE_FOR_BRANCH_TRANSACTION = '25003';
exports.PG_INAPPROPRIATE_ISOLATION_LEVEL_FOR_BRANCH_TRANSACTION = '25004';
exports.PG_NO_ACTIVE_SQL_TRANSACTION_FOR_BRANCH_TRANSACTION = '25005';
exports.PG_READ_ONLY_SQL_TRANSACTION = '25006';
exports.PG_SCHEMA_AND_DATA_STATEMENT_MIXING_NOT_SUPPORTED = '25007';
exports.PG_NO_ACTIVE_SQL_TRANSACTION = '25P01';
exports.PG_IN_FAILED_SQL_TRANSACTION = '25P02';
exports.PG_INVALID_SQL_STATEMENT_NAME = '26000';
exports.PG_TRIGGERED_DATA_CHANGE_VIOLATION = '27000';
exports.PG_INVALID_AUTHORIZATION_SPECIFICATION = '28000';
exports.PG_INVALID_PASSWORD = '28P01';
exports.PG_DEPENDENT_PRIVILEGE_DESCRIPTORS_STILL_EXIST = '2B000';
exports.PG_DEPENDENT_OBJECTS_STILL_EXIST = '2BP01';
exports.PG_INVALID_TRANSACTION_TERMINATION = '2D000';
exports.PG_SQL_ROUTINE_EXCEPTION = '2F000';
exports.PG_FUNCTION_EXECUTED_NO_RETURN_STATEMENT = '2F005';
exports.PG_SQL_ROUTINE_MODIFYING_SQL_DATA_NOT_PERMITTED = '2F002';
exports.PG_SQL_ROUTINE_PROHIBITED_SQL_STATEMENT_ATTEMPTED = '2F003';
exports.PG_SQL_ROUTINE_READING_SQL_DATA_NOT_PERMITTED = '2F004';
exports.PG_INVALID_CURSOR_NAME = '34000';
exports.PG_EXTERNAL_ROUTINE_EXCEPTION = '38000';
exports.PG_CONTAINING_SQL_NOT_PERMITTED = '38001';
exports.PG_EXTERNAL_ROUTINE_MODIFYING_SQL_DATA_NOT_PERMITTED = '38002';
exports.PG_EXTERNAL_ROUTINE_PROHIBITED_SQL_STATEMENT_ATTEMPTED = '38003';
exports.PG_EXTERNAL_ROUTINE_READING_SQL_DATA_NOT_PERMITTED = '38004';
exports.PG_EXTERNAL_ROUTINE_INVOCATION_EXCEPTION = '39000';
exports.PG_INVALID_SQLSTATE_RETURNED = '39001';
exports.PG_EXTERNAL_ROUTINE_NULL_VALUE_NOT_ALLOWED = '39004';
exports.PG_TRIGGER_PROTOCOL_VIOLATED = '39P01';
exports.PG_SRF_PROTOCOL_VIOLATED = '39P02';
exports.PG_SAVEPOINT_EXCEPTION = '3B000';
exports.PG_INVALID_SAVEPOINT_SPECIFICATION = '3B001';
exports.PG_INVALID_CATALOG_NAME = '3D000';
exports.PG_INVALID_SCHEMA_NAME = '3F000';
exports.PG_TRANSACTION_ROLLBACK = '40000';
exports.PG_TRANSACTION_INTEGRITY_CONSTRAINT_VIOLATION = '40002';
exports.PG_SERIALIZATION_FAILURE = '40001';
exports.PG_STATEMENT_COMPLETION_UNKNOWN = '40003';
exports.PG_DEADLOCK_DETECTED = '40P01';
exports.PG_SYNTAX_ERROR_OR_ACCESS_RULE_VIOLATION = '42000';
exports.PG_SYNTAX_ERROR = '42601';
exports.PG_INSUFFICIENT_PRIVILEGE = '42501';
exports.PG_CANNOT_COERCE = '42846';
exports.PG_GROUPING_ERROR = '42803';
exports.PG_WINDOWING_ERROR = '42P20';
exports.PG_INVALID_RECURSION = '42P19';
exports.PG_INVALID_FOREIGN_KEY = '42830';
exports.PG_INVALID_NAME = '42602';
exports.PG_NAME_TOO_LONG = '42622';
exports.PG_RESERVED_NAME = '42939';
exports.PG_DATATYPE_MISMATCH = '42804';
exports.PG_INDETERMINATE_DATATYPE = '42P18';
exports.PG_COLLATION_MISMATCH = '42P21';
exports.PG_INDETERMINATE_COLLATION = '42P22';
exports.PG_WRONG_OBJECT_TYPE = '42809';
exports.PG_UNDEFINED_COLUMN = '42703';
exports.PG_UNDEFINED_FUNCTION = '42883';
exports.PG_UNDEFINED_TABLE = '42P01';
exports.PG_UNDEFINED_PARAMETER = '42P02';
exports.PG_UNDEFINED_OBJECT = '42704';
exports.PG_DUPLICATE_COLUMN = '42701';
exports.PG_DUPLICATE_CURSOR = '42P03';
exports.PG_DUPLICATE_DATABASE = '42P04';
exports.PG_DUPLICATE_FUNCTION = '42723';
exports.PG_DUPLICATE_PREPARED_STATEMENT = '42P05';
exports.PG_DUPLICATE_SCHEMA = '42P06';
exports.PG_DUPLICATE_TABLE = '42P07';
exports.PG_DUPLICATE_ALIAS = '42712';
exports.PG_DUPLICATE_OBJECT = '42710';
exports.PG_AMBIGUOUS_COLUMN = '42702';
exports.PG_AMBIGUOUS_FUNCTION = '42725';
exports.PG_AMBIGUOUS_PARAMETER = '42P08';
exports.PG_AMBIGUOUS_ALIAS = '42P09';
exports.PG_INVALID_COLUMN_REFERENCE = '42P10';
exports.PG_INVALID_COLUMN_DEFINITION = '42611';
exports.PG_INVALID_CURSOR_DEFINITION = '42P11';
exports.PG_INVALID_DATABASE_DEFINITION = '42P12';
exports.PG_INVALID_FUNCTION_DEFINITION = '42P13';
exports.PG_INVALID_PREPARED_STATEMENT_DEFINITION = '42P14';
exports.PG_INVALID_SCHEMA_DEFINITION = '42P15';
exports.PG_INVALID_TABLE_DEFINITION = '42P16';
exports.PG_INVALID_OBJECT_DEFINITION = '42P17';
exports.PG_WITH_CHECK_OPTION_VIOLATION = '44000';
exports.PG_INSUFFICIENT_RESOURCES = '53000';
exports.PG_DISK_FULL = '53100';
exports.PG_OUT_OF_MEMORY = '53200';
exports.PG_TOO_MANY_CONNECTIONS = '53300';
exports.PG_CONFIGURATION_LIMIT_EXCEEDED = '53400';
exports.PG_PROGRAM_LIMIT_EXCEEDED = '54000';
exports.PG_STATEMENT_TOO_COMPLEX = '54001';
exports.PG_TOO_MANY_COLUMNS = '54011';
exports.PG_TOO_MANY_ARGUMENTS = '54023';
exports.PG_OBJECT_NOT_IN_PREREQUISITE_STATE = '55000';
exports.PG_OBJECT_IN_USE = '55006';
exports.PG_CANT_CHANGE_RUNTIME_PARAM = '55P02';
exports.PG_LOCK_NOT_AVAILABLE = '55P03';
exports.PG_OPERATOR_INTERVENTION = '57000';
exports.PG_QUERY_CANCELED = '57014';
exports.PG_ADMIN_SHUTDOWN = '57P01';
exports.PG_CRASH_SHUTDOWN = '57P02';
exports.PG_CANNOT_CONNECT_NOW = '57P03';
exports.PG_DATABASE_DROPPED = '57P04';
exports.PG_SYSTEM_ERROR = '58000';
exports.PG_IO_ERROR = '58030';
exports.PG_UNDEFINED_FILE = '58P01';
exports.PG_DUPLICATE_FILE = '58P02';
exports.PG_CONFIG_FILE_ERROR = 'F0000';
exports.PG_LOCK_FILE_EXISTS = 'F0001';
exports.PG_FDW_ERROR = 'HV000';
exports.PG_FDW_COLUMN_NAME_NOT_FOUND = 'HV005';
exports.PG_FDW_DYNAMIC_PARAMETER_VALUE_NEEDED = 'HV002';
exports.PG_FDW_FUNCTION_SEQUENCE_ERROR = 'HV010';
exports.PG_FDW_INCONSISTENT_DESCRIPTOR_INFORMATION = 'HV021';
exports.PG_FDW_INVALID_ATTRIBUTE_VALUE = 'HV024';
exports.PG_FDW_INVALID_COLUMN_NAME = 'HV007';
exports.PG_FDW_INVALID_COLUMN_NUMBER = 'HV008';
exports.PG_FDW_INVALID_DATA_TYPE = 'HV004';
exports.PG_FDW_INVALID_DATA_TYPE_DESCRIPTORS = 'HV006';
exports.PG_FDW_INVALID_DESCRIPTOR_FIELD_IDENTIFIER = 'HV091';
exports.PG_FDW_INVALID_HANDLE = 'HV00B';
exports.PG_FDW_INVALID_OPTION_INDEX = 'HV00C';
exports.PG_FDW_INVALID_OPTION_NAME = 'HV00D';
exports.PG_FDW_INVALID_STRING_LENGTH_OR_BUFFER_LENGTH = 'HV090';
exports.PG_FDW_INVALID_STRING_FORMAT = 'HV00A';
exports.PG_FDW_INVALID_USE_OF_NULL_POINTER = 'HV009';
exports.PG_FDW_TOO_MANY_HANDLES = 'HV014';
exports.PG_FDW_OUT_OF_MEMORY = 'HV001';
exports.PG_FDW_NO_SCHEMAS = 'HV00P';
exports.PG_FDW_OPTION_NAME_NOT_FOUND = 'HV00J';
exports.PG_FDW_REPLY_HANDLE = 'HV00K';
exports.PG_FDW_SCHEMA_NOT_FOUND = 'HV00Q';
exports.PG_FDW_TABLE_NOT_FOUND = 'HV00R';
exports.PG_FDW_UNABLE_TO_CREATE_EXECUTION = 'HV00L';
exports.PG_FDW_UNABLE_TO_CREATE_REPLY = 'HV00M';
exports.PG_FDW_UNABLE_TO_ESTABLISH_CONNECTION = 'HV00N';
exports.PG_PLPGSQL_ERROR = 'P0000';
exports.PG_RAISE_EXCEPTION = 'P0001';
exports.PG_NO_DATA_FOUND = 'P0002';
exports.PG_TOO_MANY_ROWS = 'P0003';
exports.PG_INTERNAL_ERROR = 'XX000';
exports.PG_DATA_CORRUPTED = 'XX001';
exports.PG_INDEX_CORRUPTED = 'XX002';
